independent-projects/arc/runtime/src/main/java/io/quarkus/arc/Arc.java

```diff
@@ -15,7 +14,7 @@
  */
 public final class Arc {
 
-    private static final AtomicReference<ArcContainerImpl> INSTANCE = new AtomicReference<>();
+    private static volatile ArcContainerImpl INSTANCE;
```

Comment:
Is there a massive advantage at dropping the `AtomicReference` in this case? Just so I know about it for the future :).